src/java.desktop/share/classes/javax/swing/text/html/CSS.java

```diff
@@ -2674,7 +2678,11 @@ Object toStyleConstants(StyleConstants key, View v) {
 
         @Override
         public int hashCode() {
-            return Float.hashCode(span);
+            int hashCode = Float.hashCode(span) | Boolean.hashCode(percentage);
+            if (units != null) {
+                hashCode |= units.hashCode();
+            }
```

Comment:
In previous cases, you used `Objects.hashCode(units)`, I presume, to avoid explicit null-check and branches. Wouldn't it work here too?